### ISSUE TITLE
Should we save running_loss and running_tar_loss also to model to resume training ?

### ISSUE THREAD

**Issue by cmdrootaccess** [NathanUA/U-2-Net#55]
Hello,

currently am able to resume training but just wanted to ask a quick question,
should i also save running_loss and running_tar_loss to my checkpoint model ? 

because when i resume training the values go back to 0.

**Comment by NathanUA** [comment]
It doesn’t matter. The code will recompute the loss before the next model weights saving after you resume the model. Instead of saving the loss, you may need to save the whole Adam optimizer status to make sure the adaptive paramteres inside Adam are consistent to your previous iteration.

**Comment by cmdrootaccess** [comment]
alright got it.

yes i am saving adam optimizer like this.

```
save_data = {
                        'epoch': epoch + 1,
                        'model_state_dict': net.state_dict(),
                        'optimizer_state_dict': optimizer.state_dict(),
                        'itr': ite_num
                    }
```
Then loading it again when resuming like this.

```
 net.load_state_dict(load_state)
 optimizer.load_state_dict(optimizer_state)
```

Am i doing it correct ?

**Comment by cmdrootaccess** [comment]
its correct , just tested. thank you
